### ISSUE TITLE
Search function returns error

### ISSUE THREAD

**Issue by gavu** [firefly-iii/firefly-iii#830]
The search function returns the following error

URL: https://my.host.name/search?q=groceries

> Error while searching
> An error occured while searching. Please check the log files for more information.

I have checked the log levels:
```
APP_ENV=production
APP_DEBUG=true
APP_FORCE_SSL=false
APP_FORCE_ROOT=
APP_KEY=[removed]
APP_LOG=daily
APP_LOG_LEVEL=debug
APP_URL=https://my.host.name/
```

The only entry in todays log is:
`[2017-09-11 10:29:26] production.ERROR: Detected invalid IBAN (""). Return NULL instead.`

Please let me know what I can do to help debug this error

Thanks,
Andrei

**Comment by JC5** [comment]
The search on the demo site works well, which leads me to believe it may be an installation error. Could you [enable debug logging](https://github.com/firefly-iii/help/wiki/Enable-debug-logging-and-debug-mode), and see what happens?

**Comment by gavu** [comment]
Debug mode is enabled, here are my logs:

```
[...]
[2017-09-11 14:30:13] production.DEBUG: Now 1 dates in set.  
[2017-09-11 14:30:13] production.DEBUG: Currentstart (2017-09-01) has become 2017-09-16.  
[2017-09-11 14:30:13] production.DEBUG: Currentstart is now 2017-09-16.  
[2017-09-11 14:30:13] production.DEBUG: Next Date match after 2017-09-16 is 2017-10-15  
[2017-09-11 14:30:13] production.DEBUG: nextExpectedMatch 2017-10-15 is after 2017-09-30, so we skip this bill now.  
[2017-09-11 14:30:13] production.DEBUG: Found dates between 2017-09-01 and 2017-09-30: ["2017-09-15 00:00:00"] 
[2017-09-11 14:30:13] production.DEBUG: Dates = 1, journalCount = 1, total = 0
[EOF]
```

- typed "groceries" in search box
- loaded the search view
- that search-loading-gif appears, after a few seconds, error is shown
- the same happens if I search directly from the search view

no other information is logged during the search, the last few lines are the one above

I do not know if it is worth mentioning. I run a https-only installation. Port 80 is blocked by the firewall. I also changed the .env flag to `APP_FORCE_SSL=true`

**Comment by JC5** [comment]
Mmm, there is clearly not enough logging present yet. That is something I must remedy.

How good are you with the console in your browser? Both Chrome and Firefox have a developer toolbar which allows you to analyse requests. Perhaps the request to `/search/search` has some error information. Otherwise, I must first extend the debug logging before I can tell you more about what might be going on.

The https-only thing does not seem to be the problem.

**Comment by gavu** [comment]
I was just about to write to you the responses I found in the browser console:

console log:
```
search?q=debt:537 GET https://budget.gavriliu.com/javascript/variables?ext=.js 
jquery-3.1.1.min.js:2 jQuery.Deferred exception: currencySymbol is not defined ReferenceError: currencySymbol is not defined
    at HTMLDocument.<anonymous> (https://budget.gavriliu.com/js/ff/firefly.js:16:22)
    at j (https://budget.gavriliu.com/js/lib/jquery-3.1.1.min.js:2:29948)
    at k (https://budget.gavriliu.com/js/lib/jquery-3.1.1.min.js:2:30262) undefined
r.Deferred.exceptionHook @ jquery-3.1.1.min.js:2
k @ jquery-3.1.1.min.js:2
setTimeout (async)
(anonymous) @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
fire @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
ready @ jquery-3.1.1.min.js:2
R @ jquery-3.1.1.min.js:3
jquery-3.1.1.min.js:2 Uncaught ReferenceError: currencySymbol is not defined
    at HTMLDocument.<anonymous> (firefly.js:16)
    at j (jquery-3.1.1.min.js:2)
    at k (jquery-3.1.1.min.js:2)
(anonymous) @ firefly.js:16
j @ jquery-3.1.1.min.js:2
k @ jquery-3.1.1.min.js:2
setTimeout (async)
r.readyException @ jquery-3.1.1.min.js:2
(anonymous) @ jquery-3.1.1.min.js:2
j @ jquery-3.1.1.min.js:2
k @ jquery-3.1.1.min.js:2
setTimeout (async)
(anonymous) @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
fire @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
k @ jquery-3.1.1.min.js:2
setTimeout (async)
(anonymous) @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
fire @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
ready @ jquery-3.1.1.min.js:2
R @ jquery-3.1.1.min.js:3
jquery-3.1.1.min.js:4 POST https://my.host.name/search/search 500 (Internal Server Error)
send @ jquery-3.1.1.min.js:4
ajax @ jquery-3.1.1.min.js:4
r.(anonymous function) @ jquery-3.1.1.min.js:4
startSearch @ index.js:21
(anonymous) @ index.js:15
j @ jquery-3.1.1.min.js:2
k @ jquery-3.1.1.min.js:2
setTimeout (async)
(anonymous) @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
fire @ jquery-3.1.1.min.js:2
i @ jquery-3.1.1.min.js:2
fireWith @ jquery-3.1.1.min.js:2
ready @ jquery-3.1.1.min.js:2
R @ jquery-3.1.1.min.js:3
```

**Comment by JC5** [comment]
```
jquery-3.1.1.min.js:4 POST https://my.host.name/search/search 500 (Internal Server Error)
```

That POST is the request that error'd. If you enable display_errors and run the search again, it might tell you what is going on.

**Comment by gavu** [comment]
you lost me on this one, how do I enable this one? 

FYI: checked the connections, there's a 404:

```
Request URL:https://my.host.name/javascript/variables?ext=.js
Request Method:GET
Status Code:404 Not Found
Remote Address:192.109.190.88:8080
Referrer Policy:no-referrer-when-downgrade
```

**Comment by gavu** [comment]
I enabled the display_errors in php.ini, restarted apache, same error, did not receive any information. Checked the response headers for `https://my.host.name/search/search` and got a token mismatch response. 

```
(1/1) TokenMismatchException
in VerifyCsrfToken.php (line 68)
at VerifyCsrfToken->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in ShareErrorsFromSession.php (line 49)
at ShareErrorsFromSession->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in StartSession.php (line 64)
at StartSession->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in AddQueuedCookiesToResponse.php (line 37)
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in EncryptCookies.php (line 59)
at EncryptCookies->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in Sandstorm.php (line 46)
at Sandstorm->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in Pipeline.php (line 102)
at Pipeline->then(object(Closure))
in Router.php (line 574)
at Router->runRouteWithinStack(object(Route), object(Request))
in Router.php (line 533)
at Router->dispatchToRoute(object(Request))
in Router.php (line 511)
at Router->dispatch(object(Request))
in Kernel.php (line 176)
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))
in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in CheckForMaintenanceMode.php (line 46)
at CheckForMaintenanceMode->handle(object(Request), object(Closure))
in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))
in Pipeline.php (line 102)
at Pipeline->then(object(Closure))
in Kernel.php (line 151)
at Kernel->sendRequestThroughRouter(object(Request))
in Kernel.php (line 116)
at Kernel->handle(object(Request))
in index.php (line 57)
(1/1) TokenMismatchException
--
in VerifyCsrfToken.php (line 68)
at VerifyCsrfToken->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in ShareErrorsFromSession.php (line 49)
at ShareErrorsFromSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in StartSession.php (line 64)
at StartSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in AddQueuedCookiesToResponse.php (line 37)
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in EncryptCookies.php (line 59)
at EncryptCookies->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Sandstorm.php (line 46)
at Sandstorm->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Router.php (line 574)
at Router->runRouteWithinStack(object(Route), object(Request))in Router.php (line 533)
at Router->dispatchToRoute(object(Request))in Router.php (line 511)
at Router->dispatch(object(Request))in Kernel.php (line 176)
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckForMaintenanceMode.php (line 46)
at CheckForMaintenanceMode->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Kernel.php (line 151)
at Kernel->sendRequestThroughRouter(object(Request))in Kernel.php (line 116)
at Kernel->handle(object(Request))in index.php (line 57)


(1/1) TokenMismatchException
--
in VerifyCsrfToken.php (line 68)
at VerifyCsrfToken->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in ShareErrorsFromSession.php (line 49)
at ShareErrorsFromSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in StartSession.php (line 64)
at StartSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in AddQueuedCookiesToResponse.php (line 37)
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in EncryptCookies.php (line 59)
at EncryptCookies->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Sandstorm.php (line 46)
at Sandstorm->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Router.php (line 574)
at Router->runRouteWithinStack(object(Route), object(Request))in Router.php (line 533)
at Router->dispatchToRoute(object(Request))in Router.php (line 511)
at Router->dispatch(object(Request))in Kernel.php (line 176)
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckForMaintenanceMode.php (line 46)
at CheckForMaintenanceMode->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Kernel.php (line 151)
at Kernel->sendRequestThroughRouter(object(Request))in Kernel.php (line 116)
at Kernel->handle(object(Request))in index.php (line 57)
```

**Comment by JC5** [comment]
Uninstall the `javascript-common` package and you should be good to go.

**Comment by gavu** [comment]
It works, thanks!
